### PR TITLE
GSGGR-150 Migrate geoshop from pip requirements to poetry

### PR DESCRIPTION
**Pull request by lanseg**
Migrated geoshop-back to poetry, along with bumping dependency versions.

To run locally, do ```docker compose up```

### CODE REVIEW

**Review by danduk82 — APPROVED** [review]

**Review by marionb — CHANGES_REQUESTED** [review]
@lanseg  Thnaks for the work

Running the composition from scratch runs in to errors when in the migrations container. My guess is that you need to add a django package to the default_settings.py 

Here is the error:
```
migrate-1  | django.core.exceptions.ImproperlyConfigured: allauth.account.middleware.AccountMiddleware must be added to settings.MIDDLEWARE
Gracefully stopping... (press Ctrl+C again to force)
service "migrate" didn't complete successfully: exit 1
```

File: Dockerfile [hunk 1]
```diff
@@ -1,20 +1,14 @@
...
 WORKDIR /app/geoshop_back/
-RUN pip3 install -r requirements.txt
-
-# Update C env vars so compiler can find gdal
-ENV CPLUS_INCLUDE_PATH=/usr/include/gdal
-ENV C_INCLUDE_PATH=/usr/include/gdal
-ENV PYTHONUNBUFFERED 1
-
 COPY . /app/geoshop_back/
-RUN mv /app/geoshop_back/default_settings.py /app/geoshop_back/settings.py
```
Comment (marionb) [inline Dockerfile:20]: This is needed because django will look for the settings.py to run.

Comment (lanseg) [reply Dockerfile:20]: Fixed. Moving only if there is no settings.py already (e.g. if user created it)

File: Dockerfile [hunk 2]
```diff
@@ -1,20 +1,14 @@
...
+
+RUN apt update && \
+    apt upgrade -y && \
+    apt install -y bash postgresql curl \
```
[the remark this answers is not in the document]
Comment (lanseg) [reply Dockerfile]: No files in the latest version

File: Dockerfile [hunk 3]
```diff
@@ -1,20 +1,14 @@
...
+                   python3 python3-poetry python3-setuptools gunicorn \
+                   libgdal-dev libffi-dev && \
```
[the remark this answers is not in the document]
Comment (lanseg) [reply Dockerfile]: Not needed now

**Review by maltaesousa — APPROVED** [review]
Thanks @lanseg for the work and @marionb for the review and tests :)

**Review by marionb — APPROVED** [review]
Thanks for the changes it is now working.

**Pull request merged**